## CI Status Update

The CI checks are showing failures, but these are due to GitHub Actions infrastructure/permissions issues, not code issues:

### ✅ Lint Check
Passes locally with no errors. CI failure is due to artifact upload permissions:
```
Error: Create Artifact Container failed: The artifact name eslint-v8-cache-key-state-Lint is not valid
```

### ✅ Tests Check  
All tests pass (4/4 including the new plugin test). CI failure is due to missing `checks: write` permission:
```
Tests:       4 passed, 4 total
##[error]Missing checks: write permission
```

### Local Verification
All checks pass locally:
- ✅ `npm run lint` - No errors
- ✅ `npm test` - 4/4 tests pass
- ✅ `npm run build` - Builds successfully

The code changes are ready for review. The CI failures are unrelated to this PR.

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -1,6 +1,10 @@
 name: Lint
 on: [pull_request]
 
+permissions:
+  checks: write
+  pull-requests: write
+
 jobs:
   eslint:
     runs-on: ubuntu-latest

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -9,12 +9,18 @@ jobs:
   eslint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Yarn
         shell: bash
         run: yarn install --immutable
 
-      - uses: Maggi64/eslint-plus-action@master
+      - name: Run lint
+        run: npm run lint:report
+        continue-on-error: true
+
+      - name: Annotate code with linting results
+        uses: ataylorme/eslint-annotate-action@v3
         with:
-          npmInstall: false
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          report-json: "eslint_report.json"

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,6 +1,10 @@
 name: Tests
 on: [pull_request]
 
+permissions:
+  checks: write
+  pull-requests: write
+
 jobs:
   tests:
     runs-on: ubuntu-latest

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,6 @@ node_modules
 yarn-error.log
 .env
 examples/app.js
+coverage
+report.json
+eslint_report.json

--- a/package.json
+++ b/package.json
@@ -27,6 +27,7 @@
     "build": "rollup -c",
     "build:examples": "cd examples && rollup -c",
     "lint": "eslint src",
+    "lint:report": "eslint \"src/**/*.ts*\" --output-file eslint_report.json --format json",
     "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js tests",
     "test:ci": "yarn test --ci --silent --coverage --json --watchAll=false --testLocationInResults --outputFile=report.json",
     "serve": "npx live-server --port=3030 --no-browser --ignore='.*,src,tests'",

